src/types/did_parser.rs

```diff
@@ -19,7 +19,7 @@ peg::parser! {
         /// let parsed = parse_ethr_did("ethr:mainnet:0xb9c5714089478a327f09197987f16f9e5d936e8a").unwrap();
         /// assert_eq!(
         ///    parsed,
-        ///    MethodAndId {
+        ///    Did {
```

Comment:
I suggest we follow the following document for naming and the parser grammar.   You can see the section 'Method Specific Identifier' provides a grammar.  

https://github.com/decentralized-identity/ethr-did-resolver/blob/master/doc/did-method-spec.md#did-method-name